snappy/hwaccess.go

```diff
@@ -53,6 +53,18 @@ func validDevice(device string) bool {
 	return false
 }
 
+func validDeviceForUdev(device string) bool {
+	inValidPrefixes := []string{"/sys/devices", "/sys/class/gpio"}
```

Comment:
For the first note, "validPrefixes" is the right one. I could have used a whitelist (actually only one path in that case /dev/...), but I decided to trust the bug description and the reason to blacklist only the two other paths. However I see now that a better name for the function would have been "invalidDeviceForUdev".

For the second note, I agree with both observations and so also the old code should be changed, namely "validDevices" method, in the same manner.
If you think that this fix is still worthy (according to what you wrote about the coming changes) and adding the first note, I would replace the blacklist with the "whitelist" []string{"/dev/"} that will avoid a more complex path sanitizer.

What do you think?